lib/ansible/modules/system/etc_hosts.py

```diff
@@ -0,0 +1,342 @@
+                sorted_aliases_old = sorted(self.lines[found]['alias'])
+
+                # Check if alias has changed
+                if sorted_aliases_new != sorted_aliases_old:
+                    self.lines[found]['alias'] = self.params['alias']
+                    self.lines[found]['line'] = self._format_line()
+
+                    changed = True
+
+        elif action == 'remove' and found is not None:
+            # If found, remove it
+            if not self.module.check_mode:
+                for idx in reversed(found):
+                    self.lines.pop(idx)
+
+            changed = True
+
+        if changed:
+            self._write_file()
+
+        return changed
+
+    def _write_file(self):
+        # Write into tmp file
+        tpm_fd, tmp_file = tempfile.mkstemp()
+
+        # Open the hosts file
+        try:
+            f = open(tmp_file, 'wb')
+        except IOError:
+            e = get_exception()
+            self.module.fail_json(
+                msg=(
+                    "Cannot open temporary hosts file %s for writing." %
+                    tmp_file),
+                details=str(e))
+
+        # Write data back into the file
+        for line in self.lines:
+            f.write(to_bytes(line['line']))
+            self.diff['after'] += line['line']
+
+        # Close the file
+        try:
+            f.close()
+        except IOError:
+            e = get_exception()
+            self.module.fail_json(
+                msg=(
+                    "Cannot close the temporal hosts file %s after writing." %
+                    tmp_file),
+                detail=str(e))
+
+        # Move the tmp file to the right location
+        self.module.atomic_move(
+            tmp_file, self.params['path'],
+            unsafe_writes=self.params['unsafe_writes'])
+
+
+def main():
+    # Module settings
+    module = AnsibleModule(
+        argument_spec=dict(
+            ip=dict(required=True),
+            hostname=dict(),
+            alias=dict(type='list'),
+            path=dict(default='/etc/hosts'),
+            state=dict(choices=['present', 'absent'], default='present')
+        ),
+        add_file_common_args=True,
+        supports_check_mode=True
+    )
+
+    # Make shorter variables
+    state = module.params['state']
+
+    if state == 'present' and module.params['hostname'] is None:
+        module.fail_json(msg="Option 'hostname' is required.")
+
+    # Read the hosts file
+    hosts = EtcHosts(module)
+
+    changed = False
+
+    # Perform action depending on the state
+    if state == 'present':
+        changed, diff = hosts.add()
+    elif state == 'absent':
+        changed, diff = hosts.remove()
+
+    # Change file attributes if needed
+    if os.path.isfile(module.params['path']):
+        file_args = module.load_file_common_arguments(module.params)
+        changed = module.set_fs_attributes_if_different(file_args, changed)
```

Comment:
Yes, it will allow to specify the [`attributes` option](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/basic.py#L799) as in the [`file` module](http://docs.ansible.com/ansible/file_module.html).